languagetool-language-modules/sv/src/main/java/org/languagetool/tokenizers/sv/SwedishWordTokenizer.java

```diff
@@ -0,0 +1,111 @@
+/* LanguageTool, a natural language style checker
+ * Copyright (C) 2008 Daniel Naber (http://www.danielnaber.de)
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+package org.languagetool.tokenizers.sv;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.StringTokenizer;
+
+import org.languagetool.tokenizers.WordTokenizer;
+
+public class SwedishWordTokenizer extends WordTokenizer {
+
+  private static final List<String> QUOTES = Arrays.asList("'", "`", "’",  "‘", "´", ":"); // Jämför vanligt citattecken "”"
+
+  private final String svTokenizingChars;
+
+  public SwedishWordTokenizer() {
+    //remove the apostrophe etc. from the standard tokenizing characters:
+    String chars = super.getTokenizingCharacters() + "_";
+    for (String quote : QUOTES) {
+      chars = chars.replace(quote, "");
+    }
+    svTokenizingChars = chars;
+  }
+
+  /**
+   * Tokenizes just like WordTokenizer with the exception for words such as
+   * "oma's" that contain an apostrophe in their middle.
+   * @param text Text to tokenize
+   * @return List of tokens
+   */
+  @Override
+  public List<String> tokenize(final String text) {
```

Comment:
This is basically a copy of the Dutch code, isn't it? We should avoid code duplication. There should be a common super class, with both the Dutch and the Swedish code extend.

Reply:
This extension of WordTokenizer is mainly based on the Dutch, yes. So what is the preferred location and naming conventions for the extension strategies you mention? I just followed the strategies I could see from earlier implementations. But It does differ from the Dutch here as well Also from the Polish, French and German. But neither of the ones having extended the base class WordTokenizer, differ very much.

Reply:
The French, German and Polish word tokenizer don't seem to duplicate code from the Dutch tokenizer, the code looks very different. To avoid duplication, you could have e.g. a `GenericWordTokenizer` with common code in `languagtool-core` and have the Dutch and Swedish one extend it.

Reply:
I removed it and its tests for now. I don't have the needed overview of all the 25 `WordTokenizer` implementations to make a generic one yet, even though I reviewed quite a lot of them. I saw only a few strands, though.